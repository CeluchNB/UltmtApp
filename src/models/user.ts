import { Realm } from '@realm/react'

export const DisplayUserSchema: Realm.ObjectSchema = {
    name: 'DisplayUser',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string',
        localGuest: 'bool?',
    },
}
