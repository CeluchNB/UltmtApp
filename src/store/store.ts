import accountReducer from './reducers/features/account/accountReducer'
import { configureStore } from '@reduxjs/toolkit'
import liveGameReducer from './reducers/features/game/liveGameReducer'
import livePointReducer from './reducers/features/point/livePointReducer'
import managedTeamReducer from './reducers/features/team/managedTeamReducer'

const store = configureStore({
    reducer: {
        account: accountReducer,
        managedTeam: managedTeamReducer,
        liveGame: liveGameReducer,
        livePoint: livePointReducer,
    },
})

export type RootState = ReturnType<typeof store.getState>
export default store
