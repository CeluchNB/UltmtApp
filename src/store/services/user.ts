import { ApiResponse } from '../../types/services'
import { CreateUserData } from '../../types/user'

export const login = async (
    username: string,
    password: string,
): Promise<ApiResponse> => {
    const response = await fetch('https://ultmt-dev.herokuapp.com/user/login', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Accept: '*/*',
        },
        body: JSON.stringify({
            email: username,
            password: password,
        }),
    })

    if (response.ok) {
        return { data: await response.json() }
    } else {
        throw new Error(await response.text())
    }
}

export const fetchProfile = async (token: string): Promise<ApiResponse> => {
    const response = await fetch('https://ultmt-dev.herokuapp.com/user/me', {
        method: 'GET',
        headers: {
            Authorization: `Bearer ${token}`,
        },
    })

    if (response.ok) {
        return { data: await response.json() }
    } else {
        throw new Error(await response.text())
    }
}

export const createAccount = async (
    profileData: CreateUserData,
): Promise<ApiResponse> => {
    console.log('making create account call')
    const response = await fetch('https://ultmt-dev.herokuapp.com/user', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Accept: '*/*',
        },
        body: JSON.stringify({ ...profileData }),
    })

    if (response.ok) {
        return { data: await response.json() }
    } else {
        throw new Error(await response.text())
    }
}
