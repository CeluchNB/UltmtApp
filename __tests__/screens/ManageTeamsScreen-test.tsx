import * as AccountReducer from '../../src/store/reducers/features/account/accountReducer'
import * as UserData from '../../src/services/data/user'
import { ManageTeamsProps } from '../../src/types/navigation'
import ManageTeamsScreen from '../../src/screens/ManageTeamsScreen'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import React from 'react'
import { User } from '../../src/types/user'
import { fetchProfileData } from '../../fixtures/data'
import store from '../../src/store/store'
import { act, fireEvent, render, waitFor } from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const navigate = jest.fn()
const addListener = jest.fn().mockReturnValue(() => {})
const setOptions = jest.fn()

const props: ManageTeamsProps = {
    navigation: {
        navigate,
        addListener,
        setOptions,
    } as any,
    route: {} as any,
}

beforeEach(async () => {
    store.dispatch(AccountReducer.setProfile(fetchProfileData))
    jest.clearAllMocks()
})

it('should match snapshot', async () => {
    const snapshot = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    ).toJSON()

    expect(snapshot).toMatchSnapshot()
})

it('should display teams correctly', async () => {
    const { queryByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    expect(
        queryByText(`@${fetchProfileData.managerTeams[0].teamname}`),
    ).not.toBeNull()

    expect(
        queryByText(`@${fetchProfileData.playerTeams[0].teamname}`),
    ).not.toBeNull()

    expect(
        queryByText(`@${fetchProfileData.playerTeams[1].teamname}`),
    ).not.toBeNull()

    expect(
        queryByText(`@${fetchProfileData.archiveTeams[0].teamname}`),
    ).not.toBeNull()
})

it('should navigate to public team screen on playing team click', async () => {
    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const team = fetchProfileData.playerTeams[0]
    const playingTeam = getByText(`@${team.teamname}`)

    fireEvent.press(playingTeam)

    expect(navigate).toHaveBeenCalledWith('PublicTeamDetails', {
        id: team._id,
    })
})

it('should navigate to managed team screen on managing team click', async () => {
    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const team = fetchProfileData.managerTeams[0]
    const managingTeam = getByText(`@${team.teamname}`)

    fireEvent.press(managingTeam)

    expect(navigate).toHaveBeenCalledWith('ManagedTeamDetails', {
        id: team._id,
    })
})

it('should navigate to public team screen', async () => {
    const { getByText } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const team = fetchProfileData.archiveTeams[0]
    const archiveTeam = getByText(`@${team.teamname}`)

    fireEvent.press(archiveTeam)

    expect(navigate).toHaveBeenCalledWith('PublicTeamDetails', {
        id: team._id,
        archive: true,
    })
})

it('should navigate to request team', async () => {
    const { getAllByTestId } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    fireEvent.press(getAllByTestId('create-button')[0])

    expect(navigate).toHaveBeenCalledWith('RequestTeam')
})

it('should navigate to create team', async () => {
    const { getAllByTestId } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    fireEvent.press(getAllByTestId('create-button')[1])

    expect(navigate).toHaveBeenCalledWith('CreateTeam')
})

it('should handle leave team', async () => {
    const leaveTeamSpy = jest
        .spyOn(UserData, 'leaveTeam')
        .mockImplementationOnce(async (_teamId: string) => {
            const newUser: User = {
                _id: 'testid',
                firstName: 'first',
                lastName: 'last',
                email: 'test@email.com',
                username: 'testuser',
                playerTeams: [
                    {
                        _id: 'id2',
                        place: 'Place2',
                        name: 'Name2',
                        teamname: 'place2name2',
                        seasonStart: '2020',
                        seasonEnd: '2020',
                    },
                    {
                        _id: 'id3',
                        place: 'Place3',
                        name: 'Name3',
                        teamname: 'place3name3',
                        seasonStart: '2021',
                        seasonEnd: '2021',
                    },
                ],
                archiveTeams: [],
                requests: ['request1', 'request2'],
                managerTeams: [
                    {
                        _id: 'id4',
                        place: 'Place4',
                        name: 'Name4',
                        teamname: 'place4name4',
                        seasonStart: '2022',
                        seasonEnd: '2022',
                    },
                ],
                openToRequests: false,
                private: false,
            }
            return newUser
        })

    const { queryByText, getAllByTestId } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    const deleteButtons = getAllByTestId('delete-button')

    fireEvent.press(deleteButtons[0])
    // Not optimal solution, see
    // ManageTeamDetailsScreen-test.tsx for further details
    await act(async () => {})

    expect(
        queryByText(`@${fetchProfileData.playerTeams[0].teamname}`),
    ).toBeNull()
    expect(leaveTeamSpy).toHaveBeenCalled()
})

it('test update on refresh', async () => {
    const spy = jest
        .spyOn(UserData, 'fetchProfile')
        .mockReturnValueOnce(
            Promise.resolve({ ...fetchProfileData, username: 'username2' }),
        )

    const { queryByText, getByTestId } = render(
        <Provider store={store}>
            <NavigationContainer>
                <ManageTeamsScreen {...props} />
            </NavigationContainer>
        </Provider>,
    )

    expect(queryByText('username2')).toBeNull()

    const scrollView = getByTestId('mt-scroll-view')
    const { refreshControl } = scrollView.props
    await act(async () => {
        refreshControl.props.onRefresh()
    })

    expect(spy).toHaveBeenCalled()
    await waitFor(async () => await queryByText('@username2'))
})
