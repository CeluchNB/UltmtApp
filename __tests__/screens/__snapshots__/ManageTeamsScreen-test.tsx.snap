// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
    }
  }
>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "#ffd600",
          ]
        }
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    testID="mt-scroll-view"
  >
    <RCTRefreshControl />
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#3183ff",
              "fontSize": 82.1917808219178,
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        Manage Teams
      </Text>
      <View
        style={
          Object {
            "alignSelf": "center",
            "width": "75%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3183ff",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Teams I Play For
            </Text>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#707070",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Place1 Name1
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  @
                  place1name1
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2019
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="delete-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ff0000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅖
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="go-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#3183ff",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅂
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#707070",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Place2 Name2
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  @
                  place2name2
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2020
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="delete-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ff0000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅖
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="go-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#3183ff",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅂
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#707070",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Place3 Name3
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  @
                  place3name3
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2021
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="delete-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#ff0000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅖
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="go-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#3183ff",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅂
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 0,
                "marginTop": 10,
                "minWidth": 64,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="more-button"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#3183ff",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  request team
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3183ff",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Teams I Manage
            </Text>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#707070",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Place4 Name4
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  @
                  place4name4
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3183ff",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  2022
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ],
                ]
              }
              testID="go-button"
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#3183ff",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰅂
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            nativeID="animatedComponent"
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 0,
                "marginTop": 10,
                "minWidth": 64,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="more-button"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#3183ff",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  create team
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3183ff",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Requests From Teams
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#3183ff",
                  "flex": 1,
                  "fontSize": 61.64383561643835,
                  "fontWeight": "bold",
                  "marginBottom": 10,
                }
              }
            >
              Requests To Teams
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
